### ISSUE TITLE
Firefox 70 disable this extension 

### ISSUE THREAD

**Issue by simonbcn** [andreicristianpetcu/google_translate_this#19]
```
Firefox 70.0.1
Google Translate this page 2.3.1
```

Firefox disables this addon because: https://blocked.cdn.mozilla.net/11fd123a-e67d-44ab-909f-b776ea2e8d0a.html
Any workaround?

**Comment by Onurtag** [comment]
I found this
https://bugzilla.mozilla.org/show_bug.cgi?id=1593243

A workaround is modifying the manifest.json and signing it yourself but why do we have to go that far for an addon that isn't on amo.

**Comment by andreicristianpetcu** [comment]
I had a few email exchanges with amo reviewers. It's sad that we cannot use this extension without hacks.

On the bright side there will be less malware shipped trough extensions. Believe me, there are lots of such cases. I will post more details about my interaction with them. It's not an ideal situation but I understand their point of view.

**Comment by nxtprogramer** [comment]
:((

**Comment by haselwarter** [comment]
This is kind of a big deal for a lot of people who need to regularly translate pages in Firefox. I have to switch to Chrome for these pages now. Is there an issue on foreign language support for Firefox anywhere?

**Comment by dessant** [comment]
> A workaround is modifying the manifest.json and signing it yourself but why do we have to go that far for an addon that isn't on amo.

That's not a solution, Mozilla does not allow you to run a perfectly safe extension such as Google Translate This in the release version of Firefox, even if it is not listed on Firefox Add-ons, and even if you sign it for personal use only. They still control a signed extension and will blocklist it the same way.

See also what happened with Page Translator: https://github.com/jeremiahlee/page-translator/issues/26

**Comment by simonbcn** [comment]
I change to [Page Translator Revised](https://addons.mozilla.org/en-US/firefox/addon/page-translator-revised/), it works very similar to this.

**Comment by grahamperrin** [comment]
Nit: blocks such as this are not specific to Firefox. 

The block is equally effective in Waterfox Classic, and so on. 

> … Is there an issue on foreign language support for Firefox anywhere?

[The sad state of language translation in Firefox](https://www.jeremiahlee.com/posts/page-translator-is-dead/) ▶ [Firefox is getting language translation and it's safer than Google Chrome's | Trusted Reviews](https://www.trustedreviews.com/news/firefox-translation-bergamot-mozilla-3948613) ▶ 

- [Bergamot](https://browser.mt/)

**Comment by dessant** [comment]
@grahamperrin, is Waterfox planning to override the blocklist for these two extensions? I think not participating in this security theater would be a great differentiator for the Waterfox project.

**Comment by jscher2000** [comment]
Userscript hosting extensions can still do what the content script needs to do. I think they cannot prevent that because there's no code review of userscripts. Therefore:

https://greasyfork.org/en/scripts/391935-inject-google-translate-widget

**Comment by grahamperrin** [comment]
@dessant https://github.com/jeremiahlee/page-translator/issues/26#issuecomment-545636643

**Comment by grahamperrin** [comment]
https://github.com/andreicristianpetcu/google_translate_this/issues/19#issuecomment-549180326

🥇

https://www.42.fr/ par exemple: 

![2019-11-05 06:11:38](https://user-images.githubusercontent.com/192271/68184272-7b47ef80-ff96-11e9-9c56-108030a01c11.png)

![2019-11-05 06:14:19](https://user-images.githubusercontent.com/192271/68184282-80a53a00-ff96-11e9-9bfb-3c782923a221.png)

![2019-11-05 06:15:10](https://user-images.githubusercontent.com/192271/68184293-8733b180-ff96-11e9-9001-ca9941d30c04.png)

Also tested with https://www.geekfaeries.fr/ et des autres.

**Comment by dessant** [comment]
@grahamperrin, I agree that disabling blocklisting in the browser is a bad idea, I was asking if Waterfox maintainers would consider overriding the block for these two specific extensions, since they're safe to use, and offer a better user experience than an user script.

**Comment by sprite-1** [comment]
@jscher2000 can you make it so that calling the function automatically loads the translation instead of having to do another click to translate?

**Comment by andreicristianpetcu** [comment]
@jscher2000 The user script will not work on sites with CSP such as GitHub. You can try to disable CSP with this extension and then run the user script https://addons.mozilla.org/en-US/firefox/addon/laboratory-by-mozilla/

Disabling CSP for every site is NOT a good idea and this extension gives a bit more control. I did a lot of fancy stuff to inject new CSP rules in my extension.

**Comment by jscher2000** [comment]
> @jscher2000 can you make it so that calling the function automatically loads the translation instead of having to do another click to translate?

If you change `autoDisplay: true` to `autoDisplay: false` on line 21, it seems Google will translate automatically, but I haven't tested it much.

**Comment by jscher2000** [comment]
> @jscher2000 The user script will not work on sites with CSP such as GitHub. You can try to disable CSP with this extension and then run the user script https://addons.mozilla.org/en-US/firefox/addon/laboratory-by-mozilla/

Yes, this is a limitation, but the best approach to solving it is not clear. Users often have at least one extension modifying CSP (such as uBlock Origin) so it can be unpredictable adding a new one to the mix. Still not solved: https://bugzilla.mozilla.org/show_bug.cgi?id=1421725

**Comment by grahamperrin** [comment]
@dessant I think, raise the Waterfox question in the Waterfox support area.  

https://redd.it/dl42c9 will be ideal, do you have a Reddit account?

**Comment by ostrolucky** [comment]
This extension can't be installed on Firefox developer edition either. Is there any instructions available how to install it in Firefox >= 70?

**Comment by 11sideways** [comment]
checked - FF 52, 55, 56, 57, 58, 60, 65, 67, 68, 69 (and all other i think) blocks this extension, altrough i started cleared FF and disable updates+sharing stats all times.
If somebody can... Please, tell me (us) step by step, how to disable checking addons, or any solution for make this addon works.

**Comment by bhaskoro-muthohar** [comment]
So there is still no fix onto this issue?

**Comment by sprite-1** [comment]
@bhaskoro-muthohar there s nothing to fix. Mozilla decided they don't like the addon and blocked it.

**Comment by bhaskoro-muthohar** [comment]
> 
> 
> @bhaskoro-muthohar there s nothing to fix. Mozilla decided they don't like the addon and blocked it.

suck. They forced us to migrate to other browsers. Do you guys have alternative add-on or migrated to chrome?

**Comment by sprite-1** [comment]
@bhaskoro-muthohar I just disabled the blocklist to continue using. It's not recommended if you have a habit of downloading new addons every time due t security but for me, I only use a few specific addons so I don't have an issue disabling it.

**Comment by ostrolucky** [comment]
Well workarounds such as disabling blocklist was what we ask for. There is literally no instructions how to workaround the limitation. We are asking what we need to change in firefox settings or files, or anything else to continue using the browser and extension. Can you share with us how did you disable the blocklist?

**Comment by sprite-1** [comment]
Just go to `about:config` and look for `extensions.blocklist.enabled` then set it to `false`

**Comment by grahamperrin** [comment]
> … `extensions.blocklist.enabled` then set it to `false`

Please don't promote this without also making the risks unmistakably clear. 

See https://github.com/jeremiahlee/page-translator/issues/26#issuecomment-550173251 – more to add, there, in a few minutes.

**Comment by sprite-1** [comment]
@grahamperrin I literally talked about it just one post above: https://github.com/andreicristianpetcu/google_translate_this/issues/19#issuecomment-557775532

**Comment by grahamperrin** [comment]
I mean, it's **really** risky …

**Comment by nl255** [comment]
Well one possible solution so that it works in Waterfox (and possibly other forks) is to modify the extension so that it no longer trips the blocklist and then simply not submit it to Mozilla to be signed at all?  Surely Mozilla wouldn't go to the trouble of updating their blacklist to block extensions that don't even run on (vanilla) Firefox at all due to never having been signed in the first place.

**Comment by grahamperrin** [comment]
https://github.com/jeremiahlee/page-translator/issues/26#issuecomment-570078850

> The block list does include non-signed extensions.

**Comment by carlosnewmusic** [comment]
Firefox Developer Edition (73.0)
![imagen](https://user-images.githubusercontent.com/19379457/72857713-c680bd00-3cbe-11ea-8b70-d3284672cfcf.png)

**Comment by carlosnewmusic** [comment]
> Nit: blocks such as this are not specific to Firefox.
> 
> The block is equally effective in Waterfox Classic, and so on.
> 
> > … Is there an issue on foreign language support for Firefox anywhere?
> 
> [The sad state of language translation in Firefox](https://www.jeremiahlee.com/posts/page-translator-is-dead/) ▶ [Firefox is getting language translation and it's safer than Google Chrome's | Trusted Reviews](https://www.trustedreviews.com/news/firefox-translation-bergamot-mozilla-3948613) ▶
> 
>     * [Bergamot](https://browser.mt/)

![imagen](https://user-images.githubusercontent.com/19379457/72886744-a5de5480-3d0a-11ea-9dd5-a014399a1bd1.png)
It already works in all firefox only I don't know what variables can be put to the translation engine :(

**Comment by carlosnewmusic** [comment]
> > Nit: blocks such as this are not specific to Firefox.
> > The block is equally effective in Waterfox Classic, and so on.
> > > … Is there an issue on foreign language support for Firefox anywhere?
> > 
> > 
> > [The sad state of language translation in Firefox](https://www.jeremiahlee.com/posts/page-translator-is-dead/) ▶ [Firefox is getting language translation and it's safer than Google Chrome's | Trusted Reviews](https://www.trustedreviews.com/news/firefox-translation-bergamot-mozilla-3948613) ▶
> > ```
> > * [Bergamot](https://browser.mt/)
> > ```
> 
> ![imagen](https://user-images.githubusercontent.com/19379457/72886744-a5de5480-3d0a-11ea-9dd5-a014399a1bd1.png)
> It already works in all firefox only I don't know what variables can be put to the translation engine :(
who wants to help the firefox translation project, would have to apply to that work:
https://careers.mozilla.org/position/gh/1666741/

**Comment by andreicristianpetcu** [comment]
have you tried https://addons.mozilla.org/en-US/firefox/addon/s3_translator/ ?

**Comment by 109isaque10** [comment]
GUYS I FOUND A WAY, just go to about:config and search extensions.blocklist.enabled and double click it to set it to false.

**Comment by grahamperrin** [comment]
@109isaque10 discussed in November 2019; please see https://github.com/andreicristianpetcu/google_translate_this/issues/19#issuecomment-557782959 above.

**Comment by grahamperrin** [comment]
https://github.com/andreicristianpetcu/google_translate_this/issues/19#issuecomment-626663193

@andreicristianpetcu apologies for the late acknowledgement; yes, thanks. I've been using it for a while. 

http://forums.mozillazine.org/viewtopic.php?p=14852940#p14852940 December 2019 identification of what might have been the bug that caused the extension to be de-listed. I did not realise it had returned to AMO until months after its reappearance; http://forums.mozillazine.org/viewtopic.php?p=14879481#p14879481
